### PR TITLE
Obj_Comb

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Comb/z_obj_comb.c [hunk 1]
```diff
@@ -32,71 +34,538 @@ const ActorInit Obj_Comb_InitVars = {
...
+            if (this->unk_1B0 > 600) {
+                this->unk_1B0 = 600;
+            }
+            func_800B9010(&this->actor, NA_SE_EV_HONEYCOMB_FALL - SFX_FLAG);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Comb/func_8098CEAC.s")
+        Actor_MoveWithGravity(&this->actor);
+        this->actor.shape.rot.x += this->unk_1AE;
+        this->actor.shape.rot.y += this->unk_1B0;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Comb/z_obj_comb.c:474]: More angles, in hex please.

File: src/overlays/actors/ovl_Obj_Comb/z_obj_comb.c [hunk 2]
```diff
@@ -32,71 +34,538 @@ const ActorInit Obj_Comb_InitVars = {
...
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Comb/func_8098D870.s")
+void ObjComb_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjComb* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Comb/func_8098D8C8.s")
+    this->unk_1B3 = (this->collider.base.acFlags & AC_HIT) != 0;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Comb/z_obj_comb.c:503]: Does this need the != 0?

**Comment by EllipticEllipsis** [comment]
Jenkins is complaining this doesn't match.

File: src/overlays/actors/ovl_Obj_Comb/z_obj_comb.c (hunk 2 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_Obj_Comb/z_obj_comb.c:503]: Yep.

File: src/overlays/actors/ovl_Obj_Comb/z_obj_comb.c [hunk 3]
```diff
@@ -32,71 +34,538 @@ const ActorInit Obj_Comb_InitVars = {
...
+                temp_v0->speedXZ = 2.0f;
+            }
+            this->unk_1B6 = 1;
+            play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+        }
+    }
+}
+
+void func_8098DA74(ObjComb* this, GlobalContext* globalCtx) {
+    s32 temp_v0 = OBJCOMB_GET_8000(&this->actor) | OBJCOMB_GET_80(&this->actor);
```
Comment (isghj5) [inline src/overlays/actors/ovl_Obj_Comb/z_obj_comb.c:318]:
```
    temp_v1 = (((s32) temp_v0 >> 0xE) & 2) | (((s32) temp_v0 >> 7) & 1);
```
Weird... if you shift params by 0xE you should be left with only the farthest left bit which is 0x8000 but its bitwise anding by two? Is this checking if a signed params is negative?

**Review by EllipticEllipsis — APPROVED** [review]

**Review by isghj5 — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**